src/framework/theme/components/radio/radio-group.component.ts

```diff
@@ -277,4 +273,12 @@ export class NbRadioGroupComponent implements AfterContentInit, OnDestroy, Contr
       });
     }
   }
+
+  protected getUpdateCheckedValue() {
+    const checkedValue = this.radios.reduce((value, item) => item.checked ? item.value : value, undefined);
+
+    if (checkedValue && checkedValue !== this.value) {
```

Comment:
Why don't find the checked option via the `find` method and then compare its `value` with `this.value`? As in your original code, if the option value property has a 'falsy' value, the group value won't be updated.